### PR TITLE
Add `Quit on Window Close` setting

### PR DESCRIPTION
**Pull request by dusansimic**
Implemented feature exit caprine on close click using checkbox in menu bar.

Closes: https://github.com/sindresorhus/caprine/issues/622

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -25,6 +25,7 @@ module.exports = new Store({
 		sidebarHidden: false,
 		autoHideMenuBar: false,
 		notificationsMuted: false,
-		hardwareAcceleration: true
+		hardwareAcceleration: true,
+		exitOnClose: false
```
Comment (sindresorhus) [inline config.js]: I would name it `quitOnWindowClose`

File: index.js [hunk 2]
```diff
@@ -201,6 +201,10 @@ function createMainWindow() {
 	win.loadURL(mainURL);
 
 	win.on('close', e => {
+		if (config.get('exitOnClose')) {
+			app.quit();
```
Comment (sindresorhus) [inline index.js]:
```js
app.quit();
return;
```

Just to be safe. `app.quit()` is not synchronous.

File: menu.js [hunk 3]
```diff
@@ -134,6 +134,14 @@ const preferencesSubmenu = [
 		click() {
 			config.set('launchMinimized', !config.get('launchMinimized'));
 		}
+	},
+	{
+		label: 'Exit on close',
```
Comment (sindresorhus) [inline menu.js]: > Quit on Window Close

**Comment by dusansimic** [comment]
Fixed it. 👍

File: config.js [hunk 4]
```diff
@@ -25,6 +25,7 @@ module.exports = new Store({
 		sidebarHidden: false,
 		autoHideMenuBar: false,
 		notificationsMuted: false,
-		hardwareAcceleration: true
+		hardwareAcceleration: true,
+		quitOnClose: false
```
Comment (sindresorhus) [inline config.js]: Read https://github.com/sindresorhus/caprine/pull/630#discussion_r228756383 again. You forgot the `window` part.

File: menu.js [hunk 5]
```diff
@@ -134,6 +134,14 @@ const preferencesSubmenu = [
 		click() {
 			config.set('launchMinimized', !config.get('launchMinimized'));
 		}
+	},
+	{
+		label: 'Quit on close',
```
Comment (sindresorhus) [inline menu.js]: Please copy-paste exactly what I commented in https://github.com/sindresorhus/caprine/pull/630#discussion_r228756416

**Comment by sindresorhus** [comment]
There's still https://github.com/sindresorhus/caprine/pull/630/files#r228756912 todo ;)

**Comment by dusansimic** [comment]
Overlooked that one too. 🤐

**Comment by sindresorhus** [comment]
You have to update the name in the places it's used too.

**Pull request merged**